test/unit/models/classes/xml/XmlEditorTest.php

```diff
@@ -46,6 +53,34 @@ public function testGetTestXml()
         $this->assertEquals($this->xmlDoc->saveToString(), $xmlString);
     }
 
+    /**
+     * @throws XmlStorageException
+     * @throws taoQtiTest_models_classes_QtiTestConverterException
+     * @throws taoQtiTest_models_classes_QtiTestServiceException
+     */
+    public function testSaveStringTest()
+    {
+        $service = new XmlEditor();
+
+        $xmlMock = '<?xml version="1.0" encoding="UTF-8"?>
+                    <assessmentTest xmlns="http://www.imsglobal.org/xsd/imsqti_v2p1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" identifier="UnitTestQtiItem" title="UnitTestQtiItem" toolName="tao" toolVersion="3.4.0-sprint130" xsi:schemaLocation="http://www.imsglobal.org/xsd/imsqti_v2p1 http://www.imsglobal.org/xsd/qti/qtiv2p1/imsqti_v2p1.xsd">
+                        <testPart identifier="testPart-1" navigationMode="linear" submissionMode="individual">
+                            <itemSessionControl maxAttempts="0" showFeedback="false" allowReview="true" showSolution="false" allowComment="false" allowSkipping="true" validateResponses="false"/>
+                            <assessmentSection identifier="assessmentSection-1" required="true" fixed="false" title="Section 1" visible="true" keepTogether="true"/>
+                        </testPart>
+                    </assessmentTest>';
+
+        $expectJsonTest = '{"qti-type":"assessmentTest","identifier":"UnitTestQtiItem","title":"UnitTestQtiItem","toolName":"tao","toolVersion":"3.4.0-sprint130","outcomeDeclarations":[],"testParts":[{"qti-type":"testPart","identifier":"testPart-1","navigationMode":0,"submissionMode":0,"preConditions":[],"branchRules":[],"itemSessionControl":{"qti-type":"itemSessionControl","maxAttempts":0,"showFeedback":false,"allowReview":true,"showSolution":false,"allowComment":false,"validateResponses":false,"allowSkipping":true},"assessmentSections":[{"qti-type":"assessmentSection","title":"Section 1","visible":true,"keepTogether":true,"rubricBlocks":[],"sectionParts":[],"identifier":"assessmentSection-1","required":true,"fixed":false,"preConditions":[],"branchRules":[]}],"testFeedbacks":[]}],"testFeedbacks":[]}';
```

Comment:
Can you create it as an array to be easier to understand? then you can use json_encode to convert it